### PR TITLE
[DMS-368] Cascade Updates

### PR DESCRIPTION
**Pull request by simpat-adam**

### CODE REVIEW

File: src/tests/EdFi.DataManagementService.Tests.E2E/Features/Resources/UpdateResourcesValidation.feature [hunk 1]
```diff
@@ -803,14 +803,13 @@ Feature: Resources "Update" Operation validations
                     "sectionIdentifier": "SECTION ABC",
                     "courseOfferingReference": {
                         "localCourseCode": "abc",
-                        "schoolId": "4003",
```
Comment (simpat-adam) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Features/Resources/UpdateResourcesValidation.feature:806]: It looks like this test previously had a flaw with `"4003"`

File: src/core/EdFi.DataManagementService.Core/Backend/UpdateCascadeHandler.cs [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+                        break;
+                }
+            }
+
+            var filterExpression = $"{firstSegment}[?({string.Join(" && ", filters)})]";
+
+            // Evaluate
+            var filterPath = JsonPath.Parse(filterExpression);
+            var pathResult = filterPath.Evaluate(returnEdFiDoc);
+            // should have at least one match
```
Comment (simpat-adam) [inline src/core/EdFi.DataManagementService.Core/Backend/UpdateCascadeHandler.cs:150]: Is it also invalid if there is more than one match?

File: src/core/EdFi.DataManagementService.Core/Backend/UpdateCascadeHandler.cs [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        var isList = referencingReferenceJsonPaths.Any(x =>
+            x != null && x.SelectNodeFromPathAs<string>("$.referenceJsonPath", _logger)!.Contains("[*]")
+        );
+
+        if (isList)
+        {
+            // When the referenceJsonPath points to a list, we must construct a filter to
+            // find the specific element(s) in the list that needs to be updated.
+            // For example, we need to turn this: "$.classPeriods[*].classPeriodReference.classPeriodName"
+            // into this: $.classPeriods[?(@.classPeriodReference.classPeriodName == "Third period edited" && @.classPeriodReference.schoolId == 123)]
```
Comment (bradbanister) [inline src/core/EdFi.DataManagementService.Core/Backend/UpdateCascadeHandler.cs:91]: Thanks for example

File: src/core/EdFi.DataManagementService.Core/Backend/UpdateCascadeHandler.cs (hunk 2 above)
Comment (bradbanister) [reply src/core/EdFi.DataManagementService.Core/Backend/UpdateCascadeHandler.cs:150]: I think more than one match is surprising, but the right thing to do is just update them all. Maybe log a warning that doc with this uuid has duplicates?

**Review by bradbanister — APPROVED** [review]

**Review by bradbanister — APPROVED** [review]

**Pull request merged**